prov/gni/src/gnix_ep.c

```diff
@@ -120,25 +298,144 @@ struct fi_ops_tagged gnix_ep_tagged_ops;
  * EP messaging API function implementations.
  ******************************************************************************/
 
-static const recv_func_t const recv_method[] = {
-					[FI_EP_MSG] = gnix_ep_recv_msg,
-					[FI_EP_RDM] = gnix_ep_recv_rdm,
-					};
-
 static ssize_t gnix_ep_recv(struct fid_ep *ep, void *buf, size_t len,
 			    void *desc, fi_addr_t src_addr, void *context)
 {
+	int ret = FI_SUCCESS;
+	int sched_req = 0;
 	struct gnix_fid_ep *ep_priv;
+	struct gnix_fid_cq *cq;
+	struct gnix_fid_av *av;
+	struct slist_entry *item, *prev;
+	struct slist *list;
+	struct gnix_fab_req *req;
+	int matched = 0;
+	ssize_t cq_len;
+	struct gnix_address *addr_ptr;
 
 	ep_priv = container_of(ep, struct gnix_fid_ep, ep_fid);
-	assert((ep_priv->type == FI_EP_RDM) || (ep_priv->type == FI_EP_MSG));
+	if (unlikely((ep_priv->type != FI_EP_RDM) &&
+				(ep_priv->type != FI_EP_MSG)))
+		return -FI_EINVAL;
 
-	return recv_method[ep_priv->type](ep,
-					  buf,
-					  len,
-					  desc,
-					  src_addr,
-					  context);
+	if (ep_priv->type == FI_EP_RDM) {
+		av = ep_priv->av;
+		assert(av != NULL);
+		if (av->type == FI_AV_TABLE) {
+			/*
```

Comment:
Should we at least assert here for now?  Might help with debugging, especially since `addr_ptr` could be used without being defined below.